src/components/SegmentedControl/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { ReactNode } from 'react';
+import PropTypes from 'prop-types';
+import RenderIf from '../RenderIf';
+import { BaseProps } from '../types';
+import {
+    StyledSegmentedControlContainer,
+    StyledSegmentedControlLabel,
+    StyledSegmentedControlOptionsWrapper,
+} from './styled';
+import SegmentedControlOption from './SegmentedControlOption';
+
+interface Option {
+    label: ReactNode;
+    value: string;
+    icon?: ReactNode;
+}
+
+interface Props extends BaseProps {
+    label?: string;
+    variant?: 'default' | 'brand';
+    value?: string;
+    options?: Option[];
+    onChange?: (event: any) => void;
+    isDisabled?: boolean;
+    isFullWidth?: boolean;
+}
+
+/**
+ * SegmentedControl component controls user interacion with a particular process.
+ */
+
+const SegmentedControl: React.FC<Props> = props => {
+    const {
+        style,
+        label,
+        variant = 'default',
+        value,
+        options = [],
+        onChange = () => {},
+        isDisabled,
+        isFullWidth,
+    } = props;
+    const [active, setActive] = React.useState(value);
```

Comment:
remember avoid states whenever possible, in this case, we don't need a state to know who is active since the source of truth will be the value prop